src/sdam/server.ts

```diff
@@ -345,9 +346,10 @@ export class Server extends TypedEventEmitter<ServerEvents> {
         operationError instanceof MongoError &&
         operationError.code === MONGODB_ERROR_CODES.Reauthenticate
       ) {
-        reauthPromise = this.pool.reauthenticate(conn).catch(error => {
+        reauthPromise = this.pool.reauthenticate(conn);
+        reauthPromise.then(undefined, error => {
           reauthPromise = null;
-          throw error;
+          squashError(error);
```

Comment:
We want to add rejection handling, but not silence the error for the await abortable line.

Reply:
This seems to be silencing the error though.  What is the correct behavior here?

Reply:
> This isn't silencing it because we are not overwriting reauthPromise

Attaching a catch handler that never throws an error to a promise still prevents the promise from ever rejecting with anything, right?

Reply:
No, the original promise is still a rejecting one, there is a new promise created by calling `.catch` that will resolve if reauthPromise rejects. This means that reauthPromise is "handled" but it doesn't mean that reauthPromise hasn't/doesn't reject.

```js
var a = Promise.reject(new Error()); 
a.catch(() => null); // take this away to turn the above into an unhandled rejection
await Promise.resolve(); // some ticks pass
await a // still rejects!
Uncaught Error
  ...
```

Reply:
Got it, thanks for clarifying!